### PR TITLE
Makefile: improvement for acrn-config

### PR DESCRIPTION
**Pull request by jsun26intel**
- generate config patch in hypervisor/Makefile;
- add acrn-config tag in version.h

Tracked-On: #3602

Signed-off-by: Victor Sun <victor.sun@intel.com>

### CODE REVIEW

**Comment by wenlingz** [comment]
@dongyaozu please help approve this PR, which is for v1.3 release, thanks

File: hypervisor/Makefile [hunk 1]
```diff
@@ -394,6 +394,7 @@ ifneq ($(BOARD_FILE)$(SCENARIO_FILE),)
 	@cat $(HV_OBJDIR)/.cfg_src_result
 	@if [ "`sed -n /successfully/p $(HV_OBJDIR)/.cfg_src_result`" = "" ]; then exit 1;	fi
 	@echo "Import hypervisor configurations from Config-xmls, configurations in source code are ignored!"
+	@sed -i 's/#define HV_CONFIG_TOOL ""/#define HV_CONFIG_TOOL " with acrn-config"/g' $(VERSION)
```
Comment (binbinwu1) [inline hypervisor/Makefile]: According to this line, the string will be added is " with acrn-config", not just "acrn-config", right?
Please align with the commit message.

File: Makefile [hunk 2]
```diff
@@ -24,57 +24,43 @@ BUILD_VERSION ?=
...
-BOARD := `sed -n '/board/p' $(BOARD_FILE) |head -1|awk -F'"' '{print $$2}'`
-SCENARIO := `sed -n '/scenario/p' $(SCENARIO_FILE) |head -1|awk -F'"' '{print $$4}'`
+override BOARD := $(shell echo `sed -n '/<acrn-config/p' $(BOARD_FILE) | sed -r 's/.*board="(.*)".*/\1/g'`)
+override SCENARIO := $(shell echo `sed -n '/<acrn-config/p' $(SCENARIO_FILE) | sed -r 's/.*scenario="(.*)".*/\1/g'`)
 
-ifeq ($(BOARD), apl-nuc)
-override BOARD := nuc6cayh
-else ifeq ($(BOARD), kbl-nuc-i7)
-override BOARD := nuc7i7dnb
+ifneq ($(BOARD)$(SCENARIO),)
```
Comment (binbinwu1) [inline Makefile]: What if the BOARD / SCENARIO is specified instead of BOARD_FILE / SCENARIO_FILE?

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/Makefile (hunk 1 above)
Comment (jsun26intel) [reply hypervisor/Makefile]: sure, will update in commit info.

File: hypervisor/Makefile [hunk 3]
```diff
@@ -366,6 +373,31 @@ install-debug: $(HV_OBJDIR)/$(HV_FILE).map $(HV_OBJDIR)/$(HV_FILE).out
...
+	fi
+	@python3 ../misc/acrn-config/board_config/board_cfg_gen.py --board $(BOARD_FILE) --scenario $(SCENARIO_FILE) > $(HV_OBJDIR)/.cfg_src_result
+	@cat $(HV_OBJDIR)/.cfg_src_result
+	@if [ "`sed -n /successfully/p $(HV_OBJDIR)/.cfg_src_result`" = "" ]; then exit 1;	fi
+	@python3 ../misc/acrn-config/scenario_config/scenario_cfg_gen.py --board $(BOARD_FILE) --scenario $(SCENARIO_FILE) > $(HV_OBJDIR)/.cfg_src_result
+	@cat $(HV_OBJDIR)/.cfg_src_result
+	@if [ "`sed -n /successfully/p $(HV_OBJDIR)/.cfg_src_result`" = "" ]; then exit 1;	fi
+	@echo "Import hypervisor configurations from Config-xmls, configurations in source code are ignored!"
+else
+	@echo "Use hypervisor configurations from source code directly."
```
Comment (binbinwu1) [inline hypervisor/Makefile]: Does the hypervisor configurations from the source code equals to arch/x86/configs/$(CONFIG_BOARD)/platform_acpi_info.h ?

Comment (jsun26intel) [reply hypervisor/Makefile]: The platform_acpi_info.h is just one of configurations.
In current code base, platform_acpi_info.h might not be exist for some $(CONFIG_BOARD). In this case, we need to copy the template from arch/x86/configs/platform_acpi_info.h as its acpi info file.

Comment (binbinwu1) [reply hypervisor/Makefile]: OK, please address the other comment and update the PR, then I can approve it. Thanks.

**Comment by wenlingz** [comment]
@binbinwu1 please help approve this PR,thanks

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**